File: NetFramework/TouchToolkit/VB/TouchToolkitDemo/GcZoomPages/MiscFeatureDemo.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports C1.Win.TouchToolKit
Imports ClickOnceDemo.C1ZoomPages.Help_Forms
Imports TouchToolKitDemo.C1ZoomPages.Help_Forms

Namespace C1ZoomPages
    Partial Public Class MiscFeatureDemo
        Inherits DemoBase
        Public Sub New()
            InitializeComponent()
            InitializeDemo()
        End Sub

        Private Sub InitializeDemo()
            Me.Title = "Misc Features"
            Me.Description = "In this page you can try a few features."

            NoneFullScreenRadioButton.Checked = True
            MiddleCenterRadioButton.Checked = True
            AddHandler layoutButton.Click, AddressOf layoutButton_Click

            TileRadioButton.Checked = True
            backcolorPreviewPanel.BackColor = SystemColors.Control
            AddHandler backgroundButton.Click, AddressOf backgroundButton_Click
            AddHandler TileRadioButton.CheckedChanged, AddressOf BackgroundImageLayoutRadioButton_CheckedChanged
            AddHandler NoneBackgroundImagRadioButton.CheckedChanged, AddressOf BackgroundImageLayoutRadioButton_CheckedChanged
            AddHandler stretchRadioButton.CheckedChanged, AddressOf BackgroundImageLayoutRadioButton_CheckedChanged
            AddHandler zoomRadioButton.CheckedChanged, AddressOf BackgroundImageLayoutRadioButton_CheckedChanged
            AddHandler centerRadioButton.CheckedChanged, AddressOf BackgroundImageLayoutRadioButton_CheckedChanged
            AddHandler bcButton.Click, AddressOf bcButton_Click
            AddHandler biButton.Click, AddressOf biButton_Click

            AddHandler keepAspectRationButton.Click, AddressOf keepAspectRationButton_Click
        End Sub

#Region "Layout"

        Private Sub layoutButton_Click(sender As Object, e As EventArgs)
            Using form As New MiscFeatureForm()
                form.Text = "Full Screen Form"
                Dim gcZoom As New C1Zoom()
                gcZoom.FullScreenMode = GetFullScreenMode()
                gcZoom.InnerPanelLayoutMode = GetInnerPanelLayoutMode()
                gcZoom.Target = form
                gcZoom.ControlBars.Add(New C1.Win.TouchToolKit.ControlBar() With {
                    .Content = GetType(ControlBarUserControl03),
                    .Opacity = 1,
                    .DockPosition = ContentAlignment.TopRight
                })
                form.ShowDialog(Me)
            End Using
        End Sub

        Private Function GetFullScreenMode() As FullScreenMode
            If NoneFullScreenRadioButton.Checked Then
                Return FullScreenMode.None
            ElseIf FullScreenRadioButton.Checked Then
                Return FullScreenMode.FullScreen
            Else
                Return FullScreenMode.FullScreenWithTaskBar
            End If
        End Function

        Private Function GetInnerPanelLayoutMode() As InnerPanelLayoutMode
            If MiddleCenterRadioButton.Checked Then
                Return InnerPanelLayoutMode.MiddleCenter
            Else
                Return InnerPanelLayoutMode.None
            End If
        End Function

        Private Sub NoneInnerPanelLayout_Click(sender As Object, e As EventArgs)
            If Not LeftLayoutRadioButton.Checked Then
                LeftLayoutRadioButton.Checked = True
            End If
        End Sub

        Private Sub MiddleCenterInnerPanelLayout_Click(sender As Object, e As EventArgs)
            If Not MiddleCenterRadioButton.Checked Then
                MiddleCenterRadioButton.Checked = True
            End If
        End Sub

#End Region

#Region "Custom Background"

        Private _gcZoomForBackground As New C1Zoom()

        Private Sub backgroundButton_Click(sender As Object, e As EventArgs)
            Using form As New MiscFeatureForm()
                form.Text = "Custom Background Form"
                _gcZoomForBackground.Target = form
                form.ShowDialog(Me)
            End Using
        End Sub

        Private Sub BackgroundImageLayoutRadioButton_CheckedChanged(sender As Object, e As EventArgs)
            If TileRadioButton.Checked Then
                _gcZoomForBackground.BackgroundImageLayout = ImageLayout.Tile
            ElseIf NoneBackgroundImagRadioButton.Checked Then
                _gcZoomForBackground.BackgroundImageLayout = ImageLayout.None
            ElseIf stretchRadioButton.Checked Then
                _gcZoomForBackground.BackgroundImageLayout = ImageLayout.Stretch
            ElseIf zoomRadioButton.Checked Then
                _gcZoomForBackground.BackgroundImageLayout = ImageLayout.Zoom
            ElseIf centerRadioButton.Checked Then
                _gcZoomForBackground.BackgroundImageLayout = ImageLayout.Center
            End If
            backcolorPreviewPanel.BackgroundImageLayout = _gcZoomForBackground.BackgroundImageLayout
        End Sub

        Private Sub bcButton_Click(sender As Object, e As EventArgs)
            Dim dialog As New ColorDialog()
            If DialogResult.OK = dialog.ShowDialog() Then
                _gcZoomForBackground.BackColor = dialog.Color
                backcolorPreviewPanel.BackColor = dialog.Color
            End If
        End Sub

        Private Sub biButton_Click(sender As Object, e As EventArgs)
            Dim dialog As New OpenFileDialog()
            dialog.Filter = "Image Files(*.BMP;*.JPG;*.GIF;*.PNG)|*.BMP;*.JPG;*.GIF;*.PNG|All files (*.*)|*.*"
            If DialogResult.OK = dialog.ShowDialog() Then
                Dim bitmap As Bitmap = Nothing
                Try
                    bitmap = New Bitmap(dialog.FileName)
                Catch generatedExceptionName As ArgumentException
                    MessageBox.Show("This is not a suitable file.")
                    Return
                End Try
                If backcolorPreviewPanel.BackgroundImage IsNot Nothing Then
                    _gcZoomForBackground.BackgroundImage = Nothing
                    backcolorPreviewPanel.BackgroundImage.Dispose()
                    backcolorPreviewPanel.BackgroundImage = Nothing
                End If
                _gcZoomForBackground.BackgroundImage = bitmap
                backcolorPreviewPanel.BackgroundImage = bitmap
            End If
        End Sub

#End Region

#Region "KeepAspectRatio"

        Private Sub keepAspectRationButton_Click(sender As Object, e As EventArgs)
            Using form As New MiscFeatureForm()
                form.Text = "KeepAspectRatio Form"
                Dim gcZoom As New C1Zoom()
                gcZoom.KeepAspectRatio = KeepAspectRatioCheckBox.Checked
				gcZoom.Target = form
				form.ShowDialog(Me)
			End Using
		End Sub

		#End Region
	End Class
End Namespace

File: Next/TouchToolkit/CS/TouchToolkitExplorer/GcZoomPages/MiscFeatureDemo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using C1.Win.TouchToolKit;
using TouchToolkitExplorer.C1ZoomPages.Help_Forms;

namespace TouchToolkitExplorer.C1ZoomPages
{
    public partial class MiscFeatureDemo : DemoBase
    {
        public MiscFeatureDemo()
        {
            InitializeComponent();
            InitializeDemo();
        }

        private void InitializeDemo()
        {
            this.Title = "Misc Features";
            this.Description = "In this page you can try a few features.";

            NoneFullScreenRadioButton.Checked = true;
            MiddleCenterRadioButton.Checked = true;
            layoutButton.Click += layoutButton_Click;

            TileRadioButton.Checked = true;
            backcolorPreviewPanel.BackColor = SystemColors.Control;
            backgroundButton.Click += backgroundButton_Click;
            TileRadioButton.CheckedChanged += BackgroundImageLayoutRadioButton_CheckedChanged;
            NoneBackgroundImagRadioButton.CheckedChanged += BackgroundImageLayoutRadioButton_CheckedChanged;
            stretchRadioButton.CheckedChanged += BackgroundImageLayoutRadioButton_CheckedChanged;
            zoomRadioButton.CheckedChanged += BackgroundImageLayoutRadioButton_CheckedChanged;
            centerRadioButton.CheckedChanged += BackgroundImageLayoutRadioButton_CheckedChanged;
            bcButton.Click+=bcButton_Click;
            biButton.Click+=biButton_Click;

            keepAspectRationButton.Click += keepAspectRationButton_Click;
        }

        #region Layout

        private void layoutButton_Click(object sender, EventArgs e)
        {
            using (MiscFeatureForm form = new MiscFeatureForm())
            {
                form.Text = "Full Screen Form";
                C1Zoom C1Zoom = new C1Zoom();
                C1Zoom.FullScreenMode = GetFullScreenMode();
                C1Zoom.InnerPanelLayoutMode = GetInnerPanelLayoutMode();
                C1Zoom.Target = form;
                C1Zoom.ControlBars.Add(new C1.Win.TouchToolKit.ControlBar() { Content = typeof(ControlBarUserControl03), Opacity = 1, DockPosition = ContentAlignment.TopRight });
                form.ShowDialog(this);
            }
        }

        private FullScreenMode GetFullScreenMode()
        {
            if (NoneFullScreenRadioButton.Checked)
            {
                return FullScreenMode.None;
            }
            else if (FullScreenRadioButton.Checked)
            {
                return FullScreenMode.FullScreen;
            }
            else
            {
                return FullScreenMode.FullScreenWithTaskBar;
            }
        }

        private InnerPanelLayoutMode GetInnerPanelLayoutMode()
        {
            if (MiddleCenterRadioButton.Checked)
            {
                return InnerPanelLayoutMode.MiddleCenter;
            }
            else
            {
                return InnerPanelLayoutMode.None;
            }
        }

        private void NoneInnerPanelLayout_Click(object sender, EventArgs e)
        {
            if (!LeftLayoutRadioButton.Checked)
            {
                LeftLayoutRadioButton.Checked = true;
            }
        }

        private void MiddleCenterInnerPanelLayout_Click(object sender, EventArgs e)
        {
            if (!MiddleCenterRadioButton.Checked)
            {
                MiddleCenterRadioButton.Checked = true;
            }
        }

        #endregion

        #region Custom Background

        C1Zoom _C1ZoomForBackground = new C1Zoom();

        private void backgroundButton_Click(object sender, EventArgs e)
        {
            using (MiscFeatureForm form = new MiscFeatureForm())
            {
                form.Text = "Custom Background Form";
                _C1ZoomForBackground.Target = form;
                form.ShowDialog(this);
            }
        }

        private void BackgroundImageLayoutRadioButton_CheckedChanged(object sender, EventArgs e)
        {
            if (TileRadioButton.Checked)
            {
                _C1ZoomForBackground.BackgroundImageLayout = ImageLayout.Tile;
            }
            else if (NoneBackgroundImagRadioButton.Checked)
            {
                _C1ZoomForBackground.BackgroundImageLayout = ImageLayout.None;
            }
            else if (stretchRadioButton.Checked)
            {
                _C1ZoomForBackground.BackgroundImageLayout = ImageLayout.Stretch;
            }
            else if (zoomRadioButton.Checked)
            {
                _C1ZoomForBackground.BackgroundImageLayout = ImageLayout.Zoom;
            }
            else if (centerRadioButton.Checked)
            {
                _C1ZoomForBackground.BackgroundImageLayout = ImageLayout.Center;
            }
            backcolorPreviewPanel.BackgroundImageLayout = _C1ZoomForBackground.BackgroundImageLayout;
        }

        private void bcButton_Click(object sender, EventArgs e)
        {
            ColorDialog dialog = new ColorDialog();
            if (DialogResult.OK == dialog.ShowDialog())
            {
                _C1ZoomForBackground.BackColor = dialog.Color;
                backcolorPreviewPanel.BackColor = dialog.Color;
            }
        }

        private void biButton_Click(object sender, EventArgs e)
        {
            OpenFileDialog dialog = new OpenFileDialog();
            dialog.Filter = "Image Files(*.BMP;*.JPG;*.GIF;*.PNG)|*.BMP;*.JPG;*.GIF;*.PNG|All files (*.*)|*.*";
            if (DialogResult.OK == dialog.ShowDialog())
            {
                Bitmap bitmap = null;
                try
                {
                    bitmap = new Bitmap(dialog.FileName);
                }
                catch (ArgumentException)
                {
                    MessageBox.Show("This is not a suitable file.");
                    return;
                }
                if (backcolorPreviewPanel.BackgroundImage != null)
                {
                    _C1ZoomForBackground.BackgroundImage = null;
                    backcolorPreviewPanel.BackgroundImage.Dispose();
                    backcolorPreviewPanel.BackgroundImage = null;
                }
                _C1ZoomForBackground.BackgroundImage = bitmap;
                backcolorPreviewPanel.BackgroundImage = bitmap;
            }
        }

        #endregion

        #region KeepAspectRatio

        private void keepAspectRationButton_Click(object sender, EventArgs e)
        {
            using (MiscFeatureForm form = new MiscFeatureForm())
            {
                form.Text = "KeepAspectRatio Form";
                C1Zoom C1Zoom = new C1Zoom();
                C1Zoom.KeepAspectRatio = KeepAspectRatioCheckBox.Checked;
                C1Zoom.Target = form;
                form.ShowDialog(this);
            }
        }

        #endregion
    }
}
